# The Sparks Foundation #GRIPJUN21

> ## Name: Saachi Gogate
> ## Task 1: Prediction using Supervised ML
>> ### Objective: Predict the percentage of a student based on the no. of study hours. 
>> ### Data: http://bit.ly/w-data

### Importing required libraries -

In [1]:
import pandas as pd
import matplotlib as plt
import seaborn as sns
import cufflinks as cf
%matplotlib inline
cf.go_offline()

### Fetching the Data -

In [2]:
df=pd.read_csv('student_scores.csv')

In [3]:
df

Hours  Scores
0     2.5      21
1     5.1      47
2     3.2      27
3     8.5      75
4     3.5      30
5     1.5      20
6     9.2      88
7     5.5      60
8     8.3      81
9     2.7      25
10    7.7      85
11    5.9      62
12    4.5      41
13    3.3      42
14    1.1      17
15    8.9      95
16    2.5      30
17    1.9      24
18    6.1      67
19    7.4      69
20    2.7      30
21    4.8      54
22    3.8      35
23    6.9      76
24    7.8      86

In [4]:
h = df[['Hours']]
h

Hours
0     2.5
1     5.1
2     3.2
3     8.5
4     3.5
5     1.5
6     9.2
7     5.5
8     8.3
9     2.7
10    7.7
11    5.9
12    4.5
13    3.3
14    1.1
15    8.9
16    2.5
17    1.9
18    6.1
19    7.4
20    2.7
21    4.8
22    3.8
23    6.9
24    7.8

In [5]:
s = df['Scores']
s

0     21
1     47
2     27
3     75
4     30
5     20
6     88
7     60
8     81
9     25
10    85
11    62
12    41
13    42
14    17
15    95
16    30
17    24
18    67
19    69
20    30
21    54
22    35
23    76
24    86
Name: Scores, dtype: int64

### Creating a cufflinks scatter plot with line of best fit to check if there is any correlation between Hours and Scores.

In [6]:
df.iplot(x='Hours', y='Scores', kind='scatter', mode='markers',size=10, xTitle='Study hours per day', yTitle='Score',
         bestfit='True', bestfit_colors=['violet'])

### Exam scores and number of hours spent studying are positively correlated.

### Training the Data -

In [7]:
from sklearn.model_selection import train_test_split

In [8]:
h_train, h_test, s_train, s_test = train_test_split(h,s, test_size=0.3, random_state=0)

### Fitting the data -

In [9]:
from sklearn.linear_model import LinearRegression

In [10]:
lr = LinearRegression()

In [11]:
lr.fit(h_train,s_train)

LinearRegression()

### Predicting score for a student who studied for 9.5 hours per day.

In [12]:
lr.predict([[9.5]])

array([95.36219891])

## =>A student who studied 9.5 hrs/day, is predicted to get a score of 95.36%.